File: 0461_hamming_distance/hamming_distance.go
package main

// Time: O(1)
// Space: O(1)
func hammingDistance(x int, y int) int {
	xor, res := x^y, 0
	for xor != 0 {
		res++
		xor &= xor - 1
	}
	return res
}

func main() {

}

File: 0461_hamming_distance/hamming_distance.py
def hamming_distance(x: int, y: int) -> int:
    """
    Time: O(1)
    Space: O(1)
    """
    xor, res = x ^ y, 0
    while xor:
        res += 1
        xor &= (xor - 1)
    return res
